test/e2e/utils.go

```diff
@@ -0,0 +1,44 @@
+package e2e
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	nmstatev1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+)
+
+func findInterface(name string, interfaces nmstatev1.State) bool {
+	//TODO: State is a interface{} we will have to convert it to json
+	//      and look for the interface name
+	return false
+}
+
+func waitInterface(namespace string, nodeName string, name string, mustExist bool) error {
+
+	f := framework.Global
+
+	return wait.Poll(5*time.Second, 50*time.Second, func() (bool, error) {
```

Comment:
this type of waiting has one serious drawback. it will wait for 5 seconds before the first check. learned the hard way

Reply:
Done.

About WaitForOperatorDeployment since we are going to use out of cluster operator for testing, it [bypass](https://github.com/operator-framework/operator-sdk/blob/master/pkg/test/e2eutil/wait_util.go#L47) all the code internally